src/System.Linq/src/System/Linq/Select.cs

```diff
@@ -170,9 +172,58 @@ public int GetCount(bool onlyIfCheap)
             {
                 return _source.Length;
             }
+
+            public IPartition<TResult> Skip(int count)
+            {
+                if (count == 0) return (IPartition<TResult>)Clone();
+                if (count >= _source.Length) return new EmptyPartition<TResult>();
+                return new SelectIPartitionIterator<TSource, TResult>(new ListPartition<TSource>(_source, count, int.MaxValue), _selector);
+            }
+
+            public IPartition<TResult> Take(int count)
+            {
+                return count >= _source.Length ? (IPartition<TResult>)Clone() : new SelectIPartitionIterator<TSource, TResult>(new ListPartition<TSource>(_source, 0, count - 1), _selector);
+            }
```

Comment:
On the flip side, it's common for selectors to allocate, in which case this reduces the number of allocations by N - 1.

A `SelectListPartition` iterator would be possible that combined the good points of this, the advantages of the specialised select iterators, and only involved a single allocation. I considered this, but thought it overkill, but considering your point on allocation, it might be worth it.